### PR TITLE
Refactor Panel

### PR DESCRIPTION
**Pull request by machour**
* Make it implement ViewContextInterface and its own render() method
* Remove the `$app` property from base class and use DI injection where possible

### CODE REVIEW

File: src/Panel.php [hunk 1]
```diff
@@ -188,4 +187,24 @@ public function isEnabled()
...
+    }
+
+    /**
+     * Renders the view specified with optional parameters.
+     * The view will be rendered using the [[view]] component.
+     * @param string $view a view name or a path alias of the view file.
+     * @param array $params the parameters (name-value pairs) that will be extracted and made available in the view file.
+     * @return string the rendering result.
+     */
+    public function render($view, $params = [])
```
Comment (samdark) [inline src/Panel.php]:
```suggestion
    public function render(string $view, array $params = []): string
```

File: src/Panel.php [hunk 2]
```diff
@@ -188,4 +187,24 @@ public function isEnabled()
     {
         return true;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getViewPath()
```
Comment (samdark) [inline src/Panel.php]: Should be type-hinted but that's something to change in the interface first.

```suggestion
    public function getViewPath(): string
```

File: src/Panels/AssetPanel.php [hunk 3]
```diff
@@ -33,23 +33,23 @@ public function getName()
      */
     public function getSummary()
```
Comment (samdark) [inline src/Panels/AssetPanel.php]:
```suggestion
    public function getSummary(): string
```

File: src/Panels/AssetPanel.php [hunk 4]
```diff
@@ -33,23 +33,23 @@ public function getName()
...
     {
-        return $this->app->view->render('panels/assets/summary', ['panel' => $this]);
+        return $this->render('panels/assets/summary', ['panel' => $this]);
     }
 
     /**
      * {@inheritdoc}
      */
     public function getDetail()
```
Comment (samdark) [inline src/Panels/AssetPanel.php]:
```suggestion
    public function getDetail(): string
```

File: src/Panel.php (hunk 2 above)
Comment (machour) [reply src/Panel.php]: will be done globally after this PR gets merged.

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**